### PR TITLE
Persist inputs so they can be re-rendered when clicking back links

### PR DESCRIPTION
**Pull request by kingarj**
## Description

There is a requirement to re-render the input given when the manual back links are clicked. Thus a number of new items have been added to the session to preserve responses to pages such as 'exemption confirmation' where we would not normally persist answers.

## Checklist:

- [x] It contains only changes required by issue (does not contain other PR)
- [ ] Includes the link to an issue (if apply)
- [ ] I have added tests to cover my changes.

### CODE REVIEW

File: app/controllers/dates_controller.rb [hunk 1]
```diff
@@ -82,12 +83,12 @@ def daily_charge
...
-      redirect_to select_daily_date_dates_path
-    else
+    unless form.valid?
       alert = I18n.t('confirmation_form.exemption')
-      redirect_back_to(daily_charge_dates_path, alert, :daily_charge)
+      return redirect_back_to(daily_charge_dates_path, alert, :daily_charge)
     end
+    SessionManipulation::SetConfirmExempt.call(session: session)
+    redirect_to select_daily_date_dates_path
   end
```
Comment (likeuwill) [inline app/controllers/dates_controller.rb]:
```
  def confirm_daily_charge
    form = ConfirmationForm.new(params['confirm-exempt'])
    if form.valid?
      SessionManipulation::SetConfirmExempt.call(session: session)
      redirect_to select_daily_date_dates_path
    else
      alert = I18n.t('confirmation_form.exemption')
      redirect_back_to(daily_charge_dates_path, alert, :daily_charge)
    end
  end
```

File: app/controllers/vehicles_controller.rb [hunk 2]
```diff
@@ -225,4 +225,9 @@ def back_button_paths
       exempt_vehicles_path
     ]
   end
+
```
Comment (likeuwill) [inline app/controllers/vehicles_controller.rb]: Please add description to method

**Comment by likeuwill** [comment]
Future test:

**Comment by likeuwill** [comment]
in `vehicles.feature`
```
  Scenario: User enters a correct vehicle's registration and choose what vehicle's details are incorrect
    Given I am on the home page
    Then I press the Start now button
      And I should be on the enter details page
    Then I enter a vehicle's registration and choose UK
      And I press the Continue
      And I should see "Are these vehicle details correct?"
    Then I press the Back link
      **Then I should see "CAS307" as vrn value
      And I press the Continue**
    Then I press the Confirm
```
in `vehicle_steps.rb` add
```
Then('I should see {string} as vrn value') do |string|
  expect(page).to have_field('vrn', with: string)
end
```
Have fun :)

File: app/controllers/dates_controller.rb (hunk 1 above)
Comment (kingarj) [reply app/controllers/dates_controller.rb]: hey @likeuwill, unfortunately this version of the function takes the class over 100 lines which makes rubocop blow up 😢 is there a separate way around this other than switching to unless?

Comment (likeuwill) [reply app/controllers/dates_controller.rb]: Hiiooo :)
You can move
```
      SessionManipulation::SetConfirmExempt.call(session: session)
      redirect_to select_daily_date_dates_path
```
to private section
for example
``` def redirect_to_select_daily_date
...
end
```

Comment (kingarj) [reply app/controllers/dates_controller.rb]: Hmm, I think private methods must still be counted in the length of the class as my line count has now jumped up to 104.

Comment (likeuwill) [reply app/controllers/dates_controller.rb]: Yeap, my bad. So to keep 100 lines, we can refactor line `222` and `223`, just write in one line
```
 SessionManipulation::CalculateTotalCharge.call(session: session, dates: dates, weekly: true)
```
and keep first version without any private methods, we are in 100/100 lines :)

**Comment by kingarj** [comment]
@likeuwill I think I've caught everything but let me know if not :)

**Pull request merged**